akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala

```diff
@@ -40,21 +39,17 @@ trait PipeToSupport {
   }
 
   final class PipeableCompletionStage[T](val future: CompletionStage[T])(implicit executionContext: ExecutionContext) {
-    def pipeTo(recipient: ActorRef)(implicit sender: ActorRef = Actor.noSender): CompletionStage[T] = {
-      future whenComplete new BiConsumer[T, Throwable] {
-        override def accept(t: T, ex: Throwable): Unit = {
-          if (t != null) recipient ! t
-          if (ex != null) recipient ! Status.Failure(ex)
-        }
-      }
+    private def pipeTo(recipient: ActorRef)(implicit sender: ActorRef = Actor.noSender): CompletionStage[T] = {
```

Comment:
keep these method private.

Reply:
I think this is an API breakage that we can't accept

Reply:
@hepin1989 do you understand my objection of why we can't change this? ok to close?

Reply:
Yes, but could we defer this to 3.0?

Reply:
> @hepin1989 do you understand my objection of why we can't change this? ok to close?

I am ok to close , but I would like to see this to be rescheduled to 3.0.

Reply:
Well, many changes would change if we would do a 3.0, but we don't have any such plans now.

Reply:
I mean there is already an issue there, as this PR referenced. Could be added to there as a backlog,other than simply close it:)

Reply:
sure, did that in the ticket